gamemodes/terrortown/gamemode/shared/sh_sprint.lua

```diff
@@ -85,24 +108,23 @@ hook.Add("Think", "TTT2PlayerSprinting", function()
 
 	for i = 1, #plys do
 		local ply = plys[i]
+		local wantsToMove = ply:KeyDown(IN_FORWARD) or ply:KeyDown(IN_BACK) or ply:KeyDown(IN_MOVERIGHT) or ply:KeyDown(IN_MOVELEFT)
 
-		if not ply.sprintTS then continue end
+		if ply.sprintProgress == 1 and (not ply.isSprinting or not wantsToMove) then continue end
+		if ply.sprintProgress == 0 and ply.isSprinting and wantsToMove then continue end
 
-		local timeElapsed = CurTime() - ply.sprintTS
 		local modifier = {1} -- Multiple hooking support
 
-		if not ply.isSprinting then
+		if not ply.isSprinting or not wantsToMove then
```

Comment:
In which case the second condition is needed? :o

Reply:
If the player isSprinting (pressing the bind for example) but standing still, we still want to regenerate stamina